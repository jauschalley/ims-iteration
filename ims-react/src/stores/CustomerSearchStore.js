import {observable, action, decorate} from "mobx";
import RequestService from '../services/RequestService';
import config from '../config';

class CustomerSearchStore {
  searchResults = null;

  searchCustomers = (formBody) => {
    RequestService.post(`${config.backendUrl}/customers`, formBody, (response) => {
      this.searchResults = response.data.customers;
    });
  }
}

decorate(CustomerSearchStore, {
  searchResults: observable,
  searchCustomers: action
});

export default CustomerSearchStore;