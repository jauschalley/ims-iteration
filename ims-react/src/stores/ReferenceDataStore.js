import {observable, decorate} from "mobx";

class ReferenceDataStore {
  referenceData = null;

  constructor(referenceData = null) {
    this.referenceData = referenceData || require('../data/lookups.json');
  }
}

decorate(ReferenceDataStore, {
  referenceData: observable,
});

export default ReferenceDataStore;