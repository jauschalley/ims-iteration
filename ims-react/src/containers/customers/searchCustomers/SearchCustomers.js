import React from 'react';
import {observer} from "mobx-react";
import {Formik, Field} from 'formik';
import {Row, Container, Col, Form, FormLabel, Button} from 'react-bootstrap';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import CwigCard from '../../../components/CwigCard';
import { tablePaginationStore } from '../../../stores/TablePaginationStore';
import TablePaginationGroup from '../../../components/TablePaginationGroup';

const mapSelectOptions = (options, labelKey, idKey) => (
  options.map((option) => (
    <option value={option[idKey]}>
      {option[labelKey]}
    </option>
  ))
);

const SearchCustomers = observer(({customerSearchStore, referenceDataStore}) => {
  let refData = referenceDataStore.referenceData;
  let page = tablePaginationStore.page;
  let rowsPerPage = tablePaginationStore.rowsPerPage;

  return(
    <Container>
      <Row><h1 class="margin-30">Search Customers</h1></Row>
      <Formik
      initialValues={{
        firstName: '',
        lastName: '',
        email: '',
        organization: '',
        state: '',
        phone: '',
        olderRecords: false
      }}
      onSubmit={(values) => customerSearchStore.searchCustomers(values)}
      >
        {formikProps =>
          <Form>
            <CwigCard>
              <Form.Row>
                <Form.Group as={Col} md="4">
                  <Form.Label>First Name</Form.Label>
                  <Form.Control placeholder="First Name" name="firstName"></Form.Control>
                </Form.Group>
                <Form.Group as={Col} md="4">
                  <Form.Label>Last Name</Form.Label>
                  <Form.Control placeholder="Last Name" name="lastName"></Form.Control>
                </Form.Group>
                <Form.Group as={Col} md="4">
                  <Form.Label>Email</Form.Label>
                  <Form.Control placeholder="Email" name="email"></Form.Control>
                </Form.Group>
              </Form.Row>
              <Form.Row>
                <Form.Group as={Col} md="4">
                  <Form.Label>Organization</Form.Label>
                  <Form.Control placeholder="Organization" name="organization"></Form.Control>
                </Form.Group>
                <Form.Group as={Col} md="4">
                <Form.Label>State</Form.Label>
                  <Form.Control as="select" name="state">
                    {mapSelectOptions(refData.states, 'state', 'stateID')}
                  </Form.Control>
                </Form.Group>
                <Form.Group as={Col} md="4">
                  <Form.Label>Phone Number</Form.Label>
                  <Form.Control placeholder="Phone Number" name="phone"></Form.Control>
                </Form.Group>
              </Form.Row>
              <Button className="search" onClick={formikProps.handleSubmit}>Search</Button>
            </CwigCard>
          </Form>
        }
      </Formik>
      {customerSearchStore.searchResults && 
       <div>
      <TableContainer component={Paper}>
          <Table aria-label="search customers table">
            <TableHead>
              <TableRow>
                <TableCell>ID</TableCell>
                <TableCell>First name</TableCell>
                <TableCell>Last Name</TableCell>
                <TableCell>Email</TableCell>
                <TableCell>Phone</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {customerSearchStore.searchResults
              .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
              .map(row => (
                <TableRow key={row.name}>
                  <TableCell>{row.customerID}</TableCell>
                  <TableCell>{row.firstName}</TableCell>
                  <TableCell>{row.lastName}</TableCell>
                  <TableCell>{row.email}</TableCell>
                  <TableCell>{row.phone}</TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </TableContainer>
        <TablePaginationGroup tableRef={customerSearchStore.searchResults}/>
        </div>
        }
    </Container>
  );
});

export default SearchCustomers;