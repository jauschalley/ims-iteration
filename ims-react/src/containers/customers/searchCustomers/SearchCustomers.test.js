import React from 'react';
import renderer from 'react-test-renderer';
import SearchCustomers from './SearchCustomers';
import referenceData from '../newInquiry/mockReferenceData';
import ReferenceDataStore from '../../../stores/ReferenceDataStore';
import CustomerSearchStore from '../../../stores/CustomerSearchStore';

describe('<SearchCustomers />', () => {
  it('renders correctly', () => {
    const referenceDataStore = new ReferenceDataStore(referenceData);
    const customerSearchStore = new CustomerSearchStore();
    const tree = renderer
      .create(
        <SearchCustomers 
          customerSearchStore={customerSearchStore}
          referenceDataStore={referenceDataStore} 
        />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});