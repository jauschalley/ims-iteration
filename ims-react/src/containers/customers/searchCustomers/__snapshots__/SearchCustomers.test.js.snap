// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchCustomers /> renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <h1
      class="margin-30"
    >
      Search Customers
    </h1>
  </div>
  <form
    className=""
  >
    <div
      className="cwig-card container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            First Name
          </label>
          <input
            name="firstName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="First Name"
            value=""
          />
        </div>
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            Last Name
          </label>
          <input
            name="lastName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Last Name"
            value=""
          />
        </div>
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            Email
          </label>
          <input
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Email"
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            Organization
          </label>
          <input
            name="organization"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Organization"
            value=""
          />
        </div>
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            State
          </label>
          <select
            name="state"
            onBlur={[Function]}
            onChange={[Function]}
            value=""
          >
            <option
              value={1}
            >
              Alabama
            </option>
            <option
              value={2}
            >
              Alaska
            </option>
            <option
              value={3}
            >
              American Samoa
            </option>
            <option
              value={4}
            >
              Arizona
            </option>
            <option
              value={5}
            >
              Arkansas
            </option>
            <option
              value={6}
            >
              California
            </option>
            <option
              value={7}
            >
              Colorado
            </option>
            <option
              value={8}
            >
              Connecticut
            </option>
            <option
              value={9}
            >
              Delaware
            </option>
            <option
              value={10}
            >
              District of Columbia
            </option>
            <option
              value={11}
            >
              Florida
            </option>
            <option
              value={12}
            >
              Georgia
            </option>
            <option
              value={13}
            >
              Guam
            </option>
            <option
              value={14}
            >
              Hawaii
            </option>
            <option
              value={15}
            >
              Idaho
            </option>
            <option
              value={16}
            >
              Illinois
            </option>
            <option
              value={17}
            >
              Indiana
            </option>
            <option
              value={18}
            >
              Iowa
            </option>
            <option
              value={19}
            >
              Kansas
            </option>
            <option
              value={20}
            >
              Kentucky
            </option>
            <option
              value={21}
            >
              Louisiana
            </option>
            <option
              value={22}
            >
              Maine
            </option>
            <option
              value={23}
            >
              Marshall Islands
            </option>
            <option
              value={24}
            >
              Maryland
            </option>
            <option
              value={25}
            >
              Massachusetts
            </option>
            <option
              value={26}
            >
              Michigan
            </option>
            <option
              value={27}
            >
              Minnesota
            </option>
            <option
              value={28}
            >
              Mississippi
            </option>
            <option
              value={29}
            >
              Missouri
            </option>
            <option
              value={30}
            >
              Montana
            </option>
            <option
              value={31}
            >
              Nebraska
            </option>
            <option
              value={32}
            >
              Nevada
            </option>
            <option
              value={33}
            >
              New Hampshire
            </option>
            <option
              value={34}
            >
              New Jersey
            </option>
            <option
              value={35}
            >
              New Mexico
            </option>
            <option
              value={36}
            >
              New York
            </option>
            <option
              value={37}
            >
              North Carolina
            </option>
            <option
              value={38}
            >
              North Dakota
            </option>
            <option
              value={39}
            >
              Northern Mariana Islands
            </option>
            <option
              value={40}
            >
              Ohio
            </option>
            <option
              value={41}
            >
              Oklahoma
            </option>
            <option
              value={42}
            >
              Oregon
            </option>
            <option
              value={43}
            >
              Palau
            </option>
            <option
              value={44}
            >
              Pennsylvania
            </option>
            <option
              value={45}
            >
              Puerto Rico
            </option>
            <option
              value={46}
            >
              Rhode Island
            </option>
            <option
              value={47}
            >
              South Carolina
            </option>
            <option
              value={48}
            >
              South Dakota
            </option>
            <option
              value={49}
            >
              Tennessee
            </option>
            <option
              value={50}
            >
              Texas
            </option>
            <option
              value={51}
            >
              Utah
            </option>
            <option
              value={52}
            >
              Vermont
            </option>
            <option
              value={53}
            >
              Virgin Islands
            </option>
            <option
              value={54}
            >
              Virginia
            </option>
            <option
              value={55}
            >
              Washington
            </option>
            <option
              value={56}
            >
              West Virginia
            </option>
            <option
              value={57}
            >
              Wisconsin
            </option>
            <option
              value={58}
            >
              Wyoming
            </option>
          </select>
        </div>
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            Phone Number
          </label>
          <input
            name="phone"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Phone Number"
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <button
          className="search btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </form>
</div>
`;
